src/compiler.rkt

```diff
@@ -333,58 +327,50 @@
        (define outhi (ival-hi output))
        
        (define out-exp
-         (match (xor (bigfloat-signbit outlo) (bigfloat-signbit outhi))
-           [#t (+   ; both bounds are positive or negative
-                   (max (abs (true-exponent outlo))
-                        (abs (true-exponent outhi)))
-                   1)]
-           [#f (+   ; tan is (-inf, +inf) or around zero
-                (max (abs (true-exponent outlo))
-                     (abs (true-exponent outhi)))
-                1
-                (get-slack))]))
+         (+ 1 (max (abs (log2-approx outlo))
+                   (abs (log2-approx outhi)))
+         (if (xor (bigfloat-signbit outlo) (bigfloat-signbit outhi))
```

Comment:
Is this going to be an issue if we have a bound like `[0, 1]` where both bounds are positive but it still contains zero?

Reply:
Ah, yeah, this case is very rare but I have a solution inside `log2-approx` - the exponent of `0.bf` will be evaluated as `(get-slack)`. So, I still will be adding slack - and ideally it will converge step by step. Despite that this case is not tracked by this condition - the slack is still in

Reply:
Ok, got it. Tricky. Would be good to at least move this kind of contains-zero logic to a function.